src/Config/TerminusConfig.php

```diff
@@ -120,48 +144,54 @@ public function getSourceName()
     }
 
     /**
-     * Get a description of where this configuration came from.
-     *
-     * @param $key
-     * @return string
+     * Return all of the keys in the Config
+     * @return array
      */
-    public function getSource($key)
+    public function keys()
     {
-        return isset($this->sources[$key]) ? $this->sources[$key] : $this->getSourceName();
+        return array_keys($this->export());
     }
 
     /**
-     * Set the source for a given configuration item.
+     * Convert the config to an array.
      *
-     * @param $key
-     * @param $source
+     * @return array
      */
-    protected function setSource($key, $source)
+    public function toArray()
```

Comment:
Robo\Config already has `export()` that does this:

https://github.com/consolidation/config/blob/master/src/ConfigInterface.php#L74

You could call through to keep the method name `toArray()`